### PR TITLE
Ncgames 4 reviews list

### PR DESCRIPTION
**Pull request by Cloud-Monkey**
Code for Task 3 and Task 4 for nc games here, the images for planning are in an img folder in the repo,
Task 4 styling of home page is incomplete but should be functional, many thx

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
+      });
+  }, []);
   return (
     <div className="App">
-      <PageHeader className="App-header" />
+      <Routes>
+        <Route
+          path="/reviews"
+          element={
+            <HomePage reviewList={reviewList} setReviewList={setReviewList} />
```
Comment (itskatherine) [inline src/App.js]: Can you also add a "/" route so when you load the page you're not met with an empty page (can lead to the same HomePage element)

File: src/apis.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  baseURL: "https://nc-games-jdro.onrender.com/api",
+});
+
+export const getReviews = () => {
+  return ncGamesApi
+    .get("/reviews")
+    .then(({ data }) => {
+      return data.review;
+    })
+    .catch((error) => {
```
Comment (itskatherine) [inline src/apis.js]: Shouldn't have a catch block here - your errors should be caught and handled in the back end 🏀

File: src/App.js [hunk 3]
```diff
@@ -1,10 +1,30 @@
...
+import { Routes, Route } from "react-router-dom";
 
 function App() {
+  const [reviewList, setReviewList] = useState([]);
+  useEffect(() => {
+    getReviews()
+      .then((response) => {
+        setReviewList(response);
+      })
+      .catch((error) => {
```
Comment (itskatherine) [inline src/App.js]: Shouldn't have catch blocks in the front end, all errors should be caught in the backend

File: src/App.js [hunk 4]
```diff
@@ -1,10 +1,30 @@
+import React, { useState, useEffect } from "react";
 import "./App.css";
-import PageHeader from "./components/PageHeader";
+import { getReviews } from "./apis";
+import HomePage from "./components/homePage";
+import { Routes, Route } from "react-router-dom";
 
 function App() {
+  const [reviewList, setReviewList] = useState([]);
```
Comment (itskatherine) [inline src/App.js:9]: Need to add a loading state and message here to show while the reviews are loading ⌛

File: src/components/ReviewCard/index.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      <div className="pic-review">
+        <img
+          src={review_img_url}
+          alt={`gameplay of ${title}`}
+          height={150}
+          width={150}
+        />
+      </div>
+      <div className="created-review">
+        <div className="review-card-created">Date Created: {formattedDate}</div>
```
Comment (itskatherine) [inline src/components/ReviewCard/index.jsx:29]: Love that you've reformatted the date to make it more readable than the jibberish that comes back from the back end. However this text is a bit small to be readable

**Review by itskatherine — CHANGES_REQUESTED** [review]
Hi Dave, your plan is off to a good start but needs a bit more added to it: 📝 
- Can you add the routes that your app will have to the diagram, and what components will be passed by that route? You’re currently missing a route for the list of reviews. For ticket 9 you’ll need a separate page for each category with a list of related reviews - so will need route/s added for this. Also can you add what route the user will go to get to the single review page?
- You have a header and avatar in each component - it would make more sense for the header to be its own component that is a child of the App

Have a look over my comments for the plan and the list of articles and then could you please resubmit this PR 🙂

File: src/components/ReviewCard/index.jsx (hunk 5 above)
Comment (Cloud-Monkey) [reply src/components/ReviewCard/index.jsx:29]: Yeah load of nonsense they give you without the format, I've made it a bit bigger now :)

File: src/App.js (hunk 1 above)
Comment (Cloud-Monkey) [reply src/App.js]: Gotcha, have added a route with this path to the same homepage element :)

File: src/apis.js (hunk 2 above)
Comment (Cloud-Monkey) [reply src/apis.js]: cool its removed now :)

File: src/App.js (hunk 3 above)
Comment (Cloud-Monkey) [reply src/App.js]: also removed this, thx :)

File: src/App.js (hunk 4 above)
Comment (Cloud-Monkey) [reply src/App.js:9]: Added a loading state now, should be good too! :)

**Comment by Cloud-Monkey** [comment]
closing to do new pull request

**Pull request closed without merging**